packages/@adobe/spectrum-css-temp/components/table/index.css

```diff
@@ -245,22 +248,59 @@ svg.spectrum-Table-sortedIcon {
     outline: 0;
   }
 
-  /* The focus ring must be on top of the first cell, so needs to be absolutely positioned. */
+  /* The focus ring should be sticky so it is visible at all horizontal scroll positions. */
   &:focus-ring {
     &:before {
+      display: inline-block;
       content: '';
-      position: absolute;
-      top: 0;
-      bottom: 0;
+      position: sticky;
       left: 0;
+      height: 100%;
       width: var(--spectrum-selectlist-border-size-key-focus);
-      z-index: 3;
+      z-index: 4;
+      margin-inline-end: calc(var(--spectrum-selectlist-border-size-key-focus) * -1)
+    }
+
+    .spectrum-Table-cellWrapper:first-of-type {
+      display: inline-block;
+    }
+  }
+
+  /* Alternative to border-left/right. Using box shadow for the row left/right since they don't take room unlike border which would cause wiggles
+   * in the hightlight case and displace the sticky indicator.
+   */
+  &.spectrum-Table-row--highlightSelection {
+    &.is-selected {
+      /* this is the box shadow for left and right border outline */
+      &:after {
+        box-shadow: inset 1px 0 0 0 var(--spectrum-global-color-blue-500), inset -1px 0 0 var(--spectrum-global-color-blue-500);
+        content: '';
+        display: block;
+        position: absolute;
+        left: 0;
+        right: 0;
+        bottom: 0;
+        top: 0;
+        z-index: 3;
+      }
```

Comment:
Using a `border` solution for the left/right brought up issues with shifts in content when the borders were added/removed on selection/deselection. Reserving the room taken by the borders to get rid of the wiggling resulted in issues with the sticky focus indicator positioning and cut offs with table cell focus ring